components/proxy/src/main/java/com/hotels/styx/routing/handlers/RouteRefLookup.java

```diff
@@ -53,15 +53,13 @@ public RoutingObject apply(StyxObjectReference route) {
 
             return routingObjectRecord
                     .map(it -> (RoutingObject) it.getRoutingObject())
-                    .orElse((liveRequest, na) -> {
-                        liveRequest.consume();
-
-                        return Eventual.of(response(NOT_FOUND)
-                                .body("Not found: " + route.name(), UTF_8)
-                                .build()
-                                .stream()
-                        );
-                    });
+                    .orElse((liveRequest, na) ->
+                            returnEarlyWithResponse(liveRequest,
```

Comment:
I would think consume() was just discarding the body instead of storing it. Is the change ok?

Reply:
Good point. `consume` simply discards each element as it is received, whereas `aggregate` will keep the content in memory until it is complete, or the byte limit is reached.

The idea to use `aggregate` only was to ensure we had a byte limit, however it does use memory unnecessarily. I will look into how to avoid this. I originally discussed with @mikkokar, but this detail was missed. Any opinions Mikko?